staging/src/k8s.io/kubectl/pkg/util/openapi/openapi.go

```diff
@@ -79,6 +83,65 @@ func (d *document) LookupResource(gvk schema.GroupVersionKind) proto.Schema {
 	return d.models.LookupModel(modelName)
 }
 
+func (d *document) GetConsumes(gvk schema.GroupVersionKind, operation string) []string {
+	for _, path := range d.doc.GetPaths().GetPath() {
+		for _, ex := range path.GetValue().GetPatch().GetVendorExtension() {
+			if ex.GetValue().GetYaml() == "" ||
+				ex.GetName() != "x-kubernetes-group-version-kind" {
+				continue
+			}
+
+			var value map[string]string
+			err := yaml.Unmarshal([]byte(ex.GetValue().GetYaml()), &value)
+			if err != nil {
+				continue
+			}
+
+			if value["group"] == gvk.Group && value["kind"] == gvk.Kind && value["version"] == gvk.Version {
+				switch operation {
+				case "GET":
+					if path.GetValue().GetGet() == nil {
+						return nil
+					}
+					return path.GetValue().GetGet().GetConsumes()
```

Comment:
You don't have to do that if since `GetConsumes` already check for nil (that comment also applies to all occurences below).
```suggestion
					return path.GetValue().GetGet().GetConsumes()
```